crates/pallet-bridges-initializer-currency-swap/src/tests.rs

```diff
@@ -0,0 +1,382 @@
+                    + EXISTENTIAL_DEPOSIT_EVM
+            );
+        });
+    })
+}
+
+#[test]
+fn initialization_idempotence() {
+    with_runtime_lock(|| {
+        let treasury = AccountInfo {
+            account: NativeTreasury::get(),
+            balance: 1450,
+        };
+        let swap_bridge_native_evm = AccountInfo {
+            account: v1::SwapBridgeNativeToEvmPot::account_id(),
+            balance: EXISTENTIAL_DEPOSIT_NATIVE,
+        };
+
+        let swap_bridge_evm_native = AccountInfo {
+            account: v1::SwapBridgeEvmToNativePot::account_id(),
+            balance: EXISTENTIAL_DEPOSIT_EVM,
+        };
+
+        let config = v1::GenesisConfig {
+            balances: pallet_balances::GenesisConfig {
+                balances: vec![
+                    treasury.into(),
+                    ALICE.into(),
+                    BOB.into(),
+                    swap_bridge_native_evm.into(),
+                ],
+            },
+            evm_balances: pallet_balances::GenesisConfig {
+                balances: vec![
+                    LION.into(),
+                    DOG.into(),
+                    CAT.into(),
+                    FISH.into(),
+                    swap_bridge_evm_native.into(),
+                ],
+            },
+            swap_bridge_native_to_evm_pot: pallet_pot::GenesisConfig {
+                initial_state: pallet_pot::InitialState::Initialized,
+            },
+            swap_bridge_evm_to_native_pot: pallet_pot::GenesisConfig {
+                initial_state: pallet_pot::InitialState::Initialized,
+            },
+            ..Default::default()
+        };
+        new_test_ext_with(config).execute_with(move || {
+            // Verify that bridges initialization has been applied at genesis.
+            assert_eq!(
+                <InitializerVersion<v1::Test>>::get(),
+                CURRENT_BRIDGES_INITIALIZER_VERSION
+            );
+            assert!(v1::EvmNativeBridgesInitializer::is_balanced().unwrap());
+            assert_eq!(
+                v1::Balances::total_balance(&v1::SwapBridgeNativeToEvmPot::account_id()),
+                LION.balance
+                    + DOG.balance
+                    + CAT.balance
+                    + FISH.balance
+                    + EXISTENTIAL_DEPOSIT_NATIVE
+            );
+            assert_eq!(
+                v1::Balances::total_balance(&NativeTreasury::get()),
+                treasury.balance - (LION.balance + DOG.balance + CAT.balance + FISH.balance)
+            );
+            assert_eq!(
+                v1::EvmBalances::total_balance(&v1::SwapBridgeEvmToNativePot::account_id(),),
+                v1::Balances::total_balance(&NativeTreasury::get())
+                    + ALICE.balance
+                    + BOB.balance
+                    + EXISTENTIAL_DEPOSIT_EVM
+            );
+
+            for _ in 0..5 {
+                v1::Balances::transfer(Some(ALICE.account).into(), BOB.account, 1).unwrap();
```

Comment:
This does not create a new account though, but it would be a worthwhile case to check

Reply:
https://github.com/humanode-network/humanode/pull/716/commits/55a9f8339c679b0334720a67bb966f28a1329c7e